Genome annotation
===============
Gene model prediction of Sapria was generated by MAKER based on multiple sources of information including RNAseq, proteoms of closely-related species, and the uniprot-swissprot database. SNAP and AUGUSTUS were used for ab initio gene prediction.
Prior to annotation, positions of repetitive elements was generated by Repeatmakser and transfered to MAKER.

Evidence-based MAKER annotation run1
------------
The initial MAKER run was based on EST data from four Rafflesiaceae species including Sapria, proteoms from close relatives (Manihot, Populus and Vitis), and uniprot-swissprot database. We especially set the maximum intron size to be 100 kb (split_hit=100000) in MAKER to accomodate the long intron in Sapria. 

MAKER control file: maker_opt.ctl1

ab initio training and MAKER annotation run2
------------
SNAP training script: snap.sh

AUGUSTUS training script: augustus.sh

MAKER control file: maker_opt.ctl2

ab initio retraining and MAKER annotation run3
------------
SNAP was retrained based on the output from run2 following the same method described in snap.sh. We then reran MAKER based on the retrained SNAP model.

MAKER control file: maker_opt.ctl3